baybe/utils/conversion.py

```diff
@@ -24,3 +30,13 @@ def fraction_to_float(value: str | float | Fraction, /) -> float:
                 f"The provided input '{value}' could not be interpreted as a fraction."
             ) from err
     return float(value)
+
+
+def nonstring_to_tuple(x: Sequence[_T], self: type, field: Attribute) -> tuple[_T, ...]:
```

Comment:
`self` as second argument? That looks incredibly weird, is this signature required in exactly this format?

Reply:
Is this due to us using `takes_self=True`?

Reply:
Yup. Agree that it's weird, that's how it is:
https://github.com/python-attrs/attrs/issues/709#issuecomment-2247109479